server/handlers/collectionHandler.js

```diff
@@ -1,54 +1,60 @@
-const Session = require('../models/Session');
+const Joi = require('joi');
 const CollectionService = require('../services/CollectionService');
+const { addShapeUrlToRegionArrayObjects } = require('../utils/helper');
+const HttpError = require('../utils/HttpError');
+
+const collectionIdQuerySchema = Joi.object({
+  collection_id: Joi.string().uuid().required(),
+}).unknown(false);
+
+const collectionPatchQuerySchema = Joi.object({
+  owner_id: Joi.string().uuid(),
+  name: Joi.string(),
+}).unknown(false);
 
 const collectionHandlerGet = async function (req, res) {
-  const { options } = req.query;
-  const session = new Session();
-  const collectionService = new CollectionService(session);
-  const ownerCollections = await collectionService.getAllByFilter(options);
-  res.status(200).json({
-    collections: ownerCollections,
-  });
+  // filters here
+  const collectionService = new CollectionService();
+  const collections = await collectionService.getCollections(req.query);
+  res.status(200).json({ collections });
 };
 
-const collectionHandlerGetCount = async function (req, res) {
-  const { filter } = req.query;
-  const session = new Session();
-  const collectionService = new CollectionService(session);
-  const ownerCollectionsCount = await collectionService.countByFilter(filter);
-  res.status(200).json({
-    count: ownerCollectionsCount,
+const collectionHandlerGetByCollectionId = async function (req, res) {
+  await collectionIdQuerySchema.validateAsync(req.params, {
+    abortEarly: false,
   });
-};
 
-const collectionHandlerGetByCollectionId = async function (req, res) {
-  const { collectionId } = req.params;
-  const session = new Session();
-  const collectionService = new CollectionService(session);
-  const collection = await collectionService.getById(collectionId);
+  const collectionService = new CollectionService();
+  const [collection] = await collectionService.getCollectionById(
+    req.params.collection_id,
+  );
+
+  if (!collection)
+    throw new HttpError(
+      404,
+      `collection with ${req.params.collection_id} not found`,
+    );
+
+  collection.regions = addShapeUrlToRegionArrayObjects(collection.regions);
+
```

Comment:
I think this is useful to add the shape URLs, but we will also store the original collection geojson in s3 I think.